### PR TITLE
FA-441 - Added projects stream to Phabricator + cleanup

### PR DESCRIPTION
**Pull request by tovbinm**
## Description

Added projects stream to Phabricator + cleanup

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

## Issue
FA-441

### CODE REVIEW

File: sources/phabricator-source/src/phabricator.ts [hunk 1]
```diff
@@ -376,4 +390,38 @@ export class Phabricator {
...
+
+    const constraints = {phids: filter.projectIds ?? []};
+    const attachments = {members: true, ancestors: true, watchers: true};
+
+    yield* this.paginate(
+      limit,
+      (after) => {
+        return this.client.project.search({
+          queryKey: 'all',
+          order: 'newest' as any,
```
Comment (cjwooo) [inline sources/phabricator-source/src/phabricator.ts:412]: I'm suprised the `as any` is needed.

File: sources/phabricator-source/src/phabricator.ts [hunk 2]
```diff
@@ -376,4 +390,38 @@ export class Phabricator {
       }
     );
   }
+
+  async *getProjects(
+    filter: {
+      projectIds?: phid[];
```
Comment (cjwooo) [inline sources/phabricator-source/src/phabricator.ts:396]: I don't see where `projectIds` is ever populated.

File: sources/phabricator-source/src/phabricator.ts (hunk 1 above)
Comment (tovbinm) [reply sources/phabricator-source/src/phabricator.ts:422]: the Condoit types are outdated

File: sources/phabricator-source/src/phabricator.ts (hunk 2 above)
Comment (tovbinm) [reply sources/phabricator-source/src/phabricator.ts:406]: added

Comment (tovbinm) [reply sources/phabricator-source/src/phabricator.ts:406]: ok, it's not used actually, but let's keep it

Comment (tovbinm) [reply sources/phabricator-source/src/phabricator.ts]: ok! I added project filter as well

**Pull request merged**